src/generator/AutoRest.Java.Azure.Fluent/Model/MethodJvaf.cs

```diff
@@ -220,14 +242,40 @@ private static MethodType GetMethodType(MethodJvaf method)
                             return MethodType.ListByResourceGroup;
                         }
                     }
-                    else if (StringComparer.OrdinalIgnoreCase.Equals(urlSplits[3], "resourceGroups"))
+                    else if (StringComparer.OrdinalIgnoreCase.Equals(urlSplits[2], "resourceGroups"))
                     {
                         return MethodType.ListByResourceGroup;
                     }
                 }
+                if (IsTopLevelResourceUrl(methodUrl))
+                {
+                    return MethodType.Get;
+                }
+            }
+            else if (method.HttpMethod == HttpMethod.Delete)
```

Comment:
Ensuring all delete functions will end up named with Delete for Top level resources.